hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/split/DefaultSplitProvider.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.source.split;
+
+import org.apache.hudi.common.util.Option;
+
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayDeque;
+import java.util.Collection;
+import java.util.Queue;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+/**
+ *  Default split provider caches pending splits in queue and return split sequentially.
+ */
+public class DefaultSplitProvider implements SplitProvider {
+  public Queue<HoodieSourceSplit> pendingSplits;
+
+  public DefaultSplitProvider() {
+    this.pendingSplits = new ArrayDeque<>();
```

Comment:
The SplitProvider interface documentation states implementations must be thread-safe, but `ArrayDeque` here is not thread-safe.

Reply:
Replaced to ConcurrentLinkedDeque.